Corona-Warn-App/src/main/java/android/print/FilePrinter.kt

```diff
@@ -0,0 +1,48 @@
+package android.print
+
+import android.os.CancellationSignal
+import android.os.ParcelFileDescriptor
+import android.print.PrintDocumentAdapter.LayoutResultCallback
+import android.print.PrintDocumentAdapter.WriteResultCallback
+import kotlinx.coroutines.suspendCancellableCoroutine
+import java.io.File
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+/**
+ * Print [PrintDocumentAdapter] content to a file
+ * Note: this file has to be in `android.print` as workaround to be able to create
+ * [LayoutResultCallback] and [WriteResultCallback]
+ */
+class FilePrinter(private val attributes: PrintAttributes) {
+    suspend fun print(printAdapter: PrintDocumentAdapter, path: File, fileName: String) =
+        suspendCancellableCoroutine<Unit> { cont ->
+            printAdapter.onLayout(
+                null,
+                attributes,
+                null,
+                object : LayoutResultCallback() {
+                    override fun onLayoutFinished(info: PrintDocumentInfo, changed: Boolean) {
+                        printAdapter.onWrite(
+                            arrayOf(PageRange.ALL_PAGES),
+                            getOutputFileDescriptor(path, fileName),
+                            CancellationSignal(),
```

Comment:
The App crashes here on my Pixel 6 without calling onWriteFinished or onWriteFailed and without throwing any exception. 
log: ```A/libc: Fatal signal 5 (SIGTRAP), code 1 (TRAP_BRKPT), fault addr 0x78711eeb74 in tid 23259 (DefaultDispatch), pid 23194 (main)```

Reply:
pull the latest changes and try again?

Reply:
Perfect, now it works